hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieHFileDataBlock.java

```diff
@@ -103,7 +104,7 @@ public HoodieLogBlockType getBlockType() {
     FSDataOutputStream ostream = new FSDataOutputStream(baos, null);
 
     HFile.Writer writer = HFile.getWriterFactory(conf, cacheConfig)
-        .withOutputStream(ostream).withFileContext(context).create();
+        .withOutputStream(ostream).withFileContext(context).withComparator(new HoodieKVComparator()).create();
```

Comment:
Lets make sure bootstrap index writing does not use this code path. again problematic

Reply:
Planning to manually test the bootstrapping case.